Eigen/src/Core/util/Memory.h

```diff
@@ -171,7 +171,7 @@ EIGEN_DEVICE_FUNC inline void* aligned_malloc(std::size_t size)
   #if (EIGEN_DEFAULT_ALIGN_BYTES==0) || EIGEN_MALLOC_ALREADY_ALIGNED
   
     #if defined(EIGEN_HIP_DEVICE_COMPILE)
-    result = aligned_malloc(size);
+    result = ::malloc(size);
```

Comment:
is this intended?

Reply:
Yes....the previous version was a recursive call, we now call the malloc from the hip runtime header (since it is a `__device__` func, instead of `std::mallloc` which is not one).

We do not seem to be exercising this path, since they previous implementation should have crashed if run

Reply:
ok